### PR TITLE
[Chore] Feature/sc 28380 - Prepare UI Library Components

### PR DESCRIPTION
**Pull request by jonycfu**
**PR checklist**

- [ X] Ensure you have added or ran the appropriate tests for your PR.
- [X ] DCO signed

**What type of PR is this?**
Refactor
**What this PR does / why we need it**:
Prep pt.1 for UI library publication 
**Which issue(s) this PR fixes**:
28380 (partial)
**Special notes for your reviewer**:
* All downstream "dumb" components have lifted the routing responsibilities to its page-level implementations (e.g. `<Link>`, `setLocation`)
image
* Moved all page-level components into `pages/..`, with the suffix name `*Page` appended to component name
image
* Where code is complicated enough, separate `*Item` components from their collection lists. (to avoid 200+ LOC)
image
* Where code is simple enough, merge `*Item` back into its collection `*List` (to avoid extra prop-drilling)
* Although this PR focuses on `CR*` components mainly, some `SR*` components were affected as a result of the "lifting up" of implementation logic to exclude embedded component routing code.

**Does this PR introduce a user-facing change?**: No

NONE

### CODE REVIEW

File: static_report/src/components/shared/Tables/TableList/CRTableListItem/CRTableListDeltaSummary.tsx [hunk 1]
```diff
@@ -5,7 +5,7 @@ import {
   FiArrowRight,
 } from 'react-icons/fi';
 
-export function CRTableListRowsSummary({
+export function CRTableListDeltaSummary({
```
[the remark this answers is not in the document]
Comment (jonycfu) [reply static_report/src/components/shared/Tables/TableList/CRTableListItem/CRTableListDeltaSummary.tsx:8]: Delta = signifies the change increment. In our case from `base` to `target`.

`Row` really doesn't say much about the kind of summary we are providing from the table list. 
image

File: static_report/src/components/shared/Tables/TableList/CRTableListItem/index.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+          <GridItem>
+            <Flex gap={10} color="gray.500">
+              <Text>Rows</Text>
+              <CRTableListDeltaSummary
+                baseCount={baseTableDatum?.row_count}
+                targetCount={targetTableDatum?.row_count}
+              />
+            </Flex>
+          </GridItem>
+          <GridItem>{children}</GridItem>
```
[the remark this answers is not in the document]
Comment (jonycfu) [reply static_report/src/components/shared/Tables/TableList/CRTableListItem/index.tsx:77]: yes

[the remark this answers is not in the document]
Comment (jonycfu) [reply static_report/src/components/shared/Tables/TableList/CRTableListItem/index.tsx:77]: in next wave of refactors (awaiting both CR+SR => shared+pages), we will address this in next stage

**Review by neighborhood999 — APPROVED** [review]

**Comment by jonycfu** [comment]
MERGE AWAIT: Once `main` branch is stable for future (don't want it included as part of upcoming release `0.8.0`

**Pull request merged**